azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java

```
@@ -5,34 +5,176 @@
         return this;
     }
 
-    /**
-     * @return the credential for specified subscription
-     */
-    public AzureTokenCredentials getCredential(String subscriptionId) {
+    public AzureEnvironment getEnvironment() {
+        return entity.getEnvironment();
+    }
+
+    public void initialize() {
+        Objects.requireNonNull(entity, "Cannot initialize from null account entity.");
+        AzureEnvironment env = MoreObjects.firstNonNull(getEnvironment(), AzureEnvironment.AZURE);
+        AzureProfile azureProfile = new AzureProfile(env);
+        Set<String> validTenantIds = new HashSet<>();
+        if (this.entity.getTenantIds() == null) {
+            this.entity.setTenantIds(AzureResourceManager.authenticate(this.credentialBuilder.provideCredentialCommon()
+                    , azureProfile).tenants().list().stream().map(Tenant::tenantId).collect(Collectors.toList()));
+
+        }
+
+        Map<String, SubscriptionEntity> subscriptionMap = new HashMap<>();
+        this.entity.getTenantIds().forEach(tenantId -> {
+            try {
+                List<SubscriptionEntity> subscriptionsOnTenant =
+                        AzureResourceManager.authenticate(this.credentialBuilder.provideCredentialForTenant(tenantId), azureProfile).subscriptions().list()
+                                .mapPage(s -> this.toSubscriptionEntity(entity.getEnvironment(), tenantId, s)).stream().collect(Collectors.toList());
+
+                for (SubscriptionEntity s : subscriptionsOnTenant) {
+                    String key = StringUtils.lowerCase(s.getId());
+                    if (subscriptionMap.putIfAbsent(key, s) != null) {
+                        validTenantIds.add(tenantId);
+                    }
+                }
+            } catch (Exception ex) {
+                // ignore AuthenticationException since on some tenants, it doesn't allow list subscriptions
+                if (!(ExceptionUtils.getRootCause(ex) instanceof AuthenticationException)) {
```

for other exception, do we need to throw them out?

---

Let it throw exception and when user reports a valid exception, we should allow it here.